daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md

```diff
@@ -7,12 +7,30 @@ aliases:
   - "/operations/components/setup-secret-store/supported-secret-stores/kubernetes-secret-store/"
 ---
 
-## Summary
+## Default Kubernetes secret store component
+When Dapr is deployed to a Kubernetes cluster, a secret store with the name `kubernetes` is automatically provisioned. This is meant to streamline the usage of the native Kubernetes secret store but generally, it is a better practice to create a component definition like the one below with a custom name. Using a custom definition decouples referencing the secret store in your code from the hosting platform (Kubernetes) keeping you code more generic and portable. Additionally, by explicitly defining a Kubernetes secret store component you can connect to a Kubernetes secret store from a local Dapr self-hosted installation. This requires a valid [`kubeconfig`](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file.
```

Comment:
@orizohar @msfussell @yaron2  I don't like that we have a feature we're telling people "it is better practice" not to use. This completely goes against the point of Dapr, which is codifying best practices. Either we make it a feature we support, or we drop it. I don't want to get us stuck in this middle ground of *it's here, but don't use it*

Reply:
This can be rephrased better. We are *not* telling people to not use the Kubernetes secret store that is enabled by default. In terms of security, its *just as* secure as a custom defined one. This is much like how we default to the platform's DNS resolution: on k8s we use k8s name resolution and self-hosted has mDNS.

If users want to use a default store with no added maintenance of keeping a custom definition file, they can. If they need to explicitly support multiple secret stores in multiple platforms, they will create custom definition files. This really falls under the sane defaults definition for me.

Reply:
@berndverst thoughts on what @yaron2 is suggesting? Can we consider pivoting this message around user scenarios for when to use a certain approach?